### PR TITLE
Add custom palettes to draw image with demo

### PR DESCRIPTION
**Pull request by matsondawson**
For those people who don't like rainbows.  I chose to make the custom palette an image as you could load a palette from and image file, or create a palette using image ops e.g. put some constants color in then use a blur to smooth it.
I do wish there was a better way of assigning the raw palette data to the image, any ideas?

### CODE REVIEW

File: src/omv/py/py_helper.c [hunk 1]
```diff
@@ -67,6 +67,12 @@ image_t *py_helper_keyword_to_image_mutable_mask(uint n_args, const mp_obj_t *ar
     return py_helper_keyword_to_image_mutable(n_args, args, arg_index, kw_args, MP_OBJ_NEW_QSTR(MP_QSTR_mask), NULL);
 }
 
+image_t *py_helper_keyword_to_image_mutable_color_palette(uint n_args, const mp_obj_t *args, uint arg_index,
+                                                 mp_map_t *kw_args)
```
Comment (kwagyeman) [inline src/omv/py/py_helper.c:71]: Fix alignment.

File: src/omv/py/py_helper.c [hunk 2]
```diff
@@ -124,6 +130,21 @@ int py_helper_keyword_int(uint n_args, const mp_obj_t *args, uint arg_index,
...
+    if (kw_arg) {
+        return mp_obj_get_int_maybe(kw_arg->value, value);
+    } else if (n_args > arg_index) {
+        return mp_obj_get_int_maybe(args[arg_index], value);
+    }
+
+    return false;
+}
+
+
```
Comment (kwagyeman) [inline src/omv/py/py_helper.c:147]: Extra space.

File: src/omv/py/py_helper.c [hunk 3]
```diff
@@ -124,6 +130,21 @@ int py_helper_keyword_int(uint n_args, const mp_obj_t *args, uint arg_index,
     return default_val;
 }
 
+int py_helper_keyword_int_maybe(uint n_args, const mp_obj_t *args, uint arg_index,
+                          mp_map_t *kw_args, mp_obj_t kw, int* value)
```
Comment (kwagyeman) [inline src/omv/py/py_helper.c:134]: Fix alginment.

File: src/omv/py/py_helper.h [hunk 4]
```diff
@@ -20,13 +20,17 @@ image_t *py_helper_arg_to_image_color(const mp_obj_t arg);
 image_t *py_helper_keyword_to_image_mutable(uint n_args, const mp_obj_t *args, uint arg_index,
                                             mp_map_t *kw_args, mp_obj_t kw, image_t *default_val);
 image_t *py_helper_keyword_to_image_mutable_mask(uint n_args, const mp_obj_t *args, uint arg_index,
-                                                 mp_map_t *kw_args);
+                                            mp_map_t *kw_args);
+image_t *py_helper_keyword_to_image_mutable_color_palette(uint n_args, const mp_obj_t *args, uint arg_index,
```
Comment (kwagyeman) [inline src/omv/py/py_helper.h:24]: Fix alignment.

File: src/omv/py/py_helper.h [hunk 5]
```diff
@@ -20,13 +20,17 @@ image_t *py_helper_arg_to_image_color(const mp_obj_t arg);
...
+                                            mp_map_t *kw_args);
 void py_helper_keyword_rectangle(image_t *img, uint n_args, const mp_obj_t *args, uint arg_index,
                                  mp_map_t *kw_args, mp_obj_t kw, rectangle_t *r);
 void py_helper_keyword_rectangle_roi(image_t *img, uint n_args, const mp_obj_t *args, uint arg_index,
                                      mp_map_t *kw_args, rectangle_t *r);
 int py_helper_keyword_int(uint n_args, const mp_obj_t *args, uint arg_index,
                           mp_map_t *kw_args, mp_obj_t kw, int default_val);
+int py_helper_keyword_int_maybe(uint n_args, const mp_obj_t *args, uint arg_index,
+                            mp_map_t *kw_args, mp_obj_t kw, int* value);
```
Comment (kwagyeman) [inline src/omv/py/py_helper.h:33]: Fix alignment.

File: src/omv/py/py_image.c [hunk 6]
```diff
@@ -1946,6 +1945,18 @@ STATIC mp_obj_t py_image_draw_image(uint n_args, const mp_obj_t *args, mp_map_t
 
         if (arg_other->bpp != IMAGE_BPP_GRAYSCALE) nlr_raise(mp_obj_new_exception_msg(&mp_type_ValueError, "Can only specify color palette when passing a grayscale image!"));
     }
+    else {
```
Comment (kwagyeman) [inline src/omv/py/py_image.c:1948]: else on same line as }

File: src/omv/py/py_image.c [hunk 7]
```diff
@@ -1931,11 +1931,10 @@ STATIC mp_obj_t py_image_draw_image(uint n_args, const mp_obj_t *args, mp_map_t
     PY_ASSERT_TRUE_MSG((0 <= arg_alpha) && (arg_alpha <= 256), "Error: 0 <= alpha <= 256!");
     image_t *arg_msk =
         py_helper_keyword_to_image_mutable_mask(n_args, args, offset + 3, kw_args);
-    int palette = py_helper_keyword_int(n_args, args, offset + 4, kw_args, MP_OBJ_NEW_QSTR(MP_QSTR_color_palette), -1);
-
+    
     const uint16_t *color_palette = NULL;
-
-    if (palette != -1) {
+    int palette;
```
Comment (kwagyeman) [inline src/omv/py/py_image.c:1936]: add new line after

File: src/omv/py/py_image.c [hunk 8]
```diff
@@ -1946,6 +1945,18 @@ STATIC mp_obj_t py_image_draw_image(uint n_args, const mp_obj_t *args, mp_map_t
 
         if (arg_other->bpp != IMAGE_BPP_GRAYSCALE) nlr_raise(mp_obj_new_exception_msg(&mp_type_ValueError, "Can only specify color palette when passing a grayscale image!"));
     }
+    else {
+        image_t *arg_palette = py_helper_keyword_to_image_mutable_color_palette(n_args, args, offset + 4, kw_args);
```
Comment (kwagyeman) [inline src/omv/py/py_image.c:1949]: new line after

**Comment by kwagyeman** [comment]
What you did is the most space efficient way for a color palette. Because it's an image you can control it well.

**Comment by matsondawson** [comment]
All review changes are applied.

**Pull request merged**